pyproject.toml

```diff
@@ -1,12 +1,137 @@
+[build-system]
+requires = ["setuptools >= 61.0"]
+build-backend = "setuptools.build_meta"
```

Comment:
can you revert the changes here and to install_python_dependencies? I want to keep the diff small on this PR in case it makes sense to revert (if the locking actually becomes a problem)

Reply:
Had to make some change to get the poetry pyproject.toml to work with uv, but it's a smaller diff than this was